src/OpenSSL/crypto.py

```diff
@@ -3073,8 +3072,8 @@ def load_pkcs12(buffer, passphrase=None):
 
     pycacerts = []
     for i in range(_lib.sk_X509_num(cacerts)):
-        pycacert = X509.__new__(X509)
-        pycacert._x509 = _lib.sk_X509_value(cacerts, i)
+        x509 = _lib.sk_X509_value(cacerts, i)
+        pycacert = X509._from_raw_x509_ptr(x509)
```

Comment:
I'd like a double check here. I'm not confident this is right. Does `sk_X509_free` free the underlying `X509 *` in the stack? If so then this is wrong, but that would also mean the current code in master is just a giant UAF.

Reply:
`sk_x509_free` only frees the stack itself, not any of the things it contains, afaik

Reply:
Okay, if that's true (which is what I thought but it's amazing how quickly you start second guessing yourself when fixing memory bugs) then should be right.

Reply:
So the memory approach here is that each `X509` object is responsible for hte `X509*` and the `cacerts` var is only responsible for freeing the `stack`?

Reply:
Correct